src/Core/src/ImageSources/FileImageSourceService/FileImageSourceService.Android.cs

```diff
@@ -12,6 +12,37 @@ namespace Microsoft.Maui
 {
 	public partial class FileImageSourceService
 	{
+		public override async Task<IImageSourceServiceResult<Drawable>?> LoadDrawableAsync(IImageSource imageSource, Android.Widget.ImageView imageView, CancellationToken cancellationToken = default)
+		{
+			if (imageSource is IFileImageSource fileImageSource)
+			{
+				try
+				{
+					var listener = new RequestBuilderExtensions.RequestCompleteListener();
+					var glide = Glide.With(imageView.Context);
+					var builder = glide
+						.Load(fileImageSource.File, imageView.Context!)
+						.AddListener(listener);
```

Comment:
I should add the benchmarks from here:

https://github.com/dotnet/maui/compare/main...jonathanpeppers:ImageBenchmarks

The problem I was seeing was that `ImageView.Set*` was so much faster than Glide:

| Method | Mean | Error | StdDev | Gen 0 | Allocated |
|------------------- |-----------:|----------:|-----------:|-------:|----------:|
| SetImageResource | 7.631 µs | 0.0165 µs | 0.0146 µs | - | - |
| SetImageDrawable | 55.096 µs | 1.2665 µs | 3.6135 µs | 0.1221 | 736 B |
| GlideWithTarget | 209.044 µs | 4.3352 µs | 12.2275 µs | 0.2441 | 1,544 B |

It could be there is just way more interop from C# to Java? The example above would many extra JNI calls.

We would potentially have to write the Glide code in Java so we can do a single interop call?

Reply:
Yeah that was my thought too...  we really just want to throw the filename/url/font info/stream over the fence to java, in a single interop call and then wait for the callback.

Reply:
@jonathanpeppers I moved most of the logic into the android aar project... Would you be able to test the timings here again?